catkin_tools_document/doxygen.py

```diff
@@ -64,12 +64,13 @@ def generate_doxygen_config(logger, event_queue, conf, package, recursive_build_
         'HTML_HEADER': header_filename,
         'HTML_OUTPUT': output_dir,
         'IMAGE_PATH': conf.get('image_path', source_path),
-        'INPUT': source_path,
+        'INPUT': source_path + " " + conf.get('use_mdfile_as_mainpage', ""),
```

Comment:
Why does this need to be here?

Reply:
According to https://stackoverflow.com/a/13442157 the mdfile should be in the INPUT. I can't find this in the Doxygen documentation, but I experienced it not working without the mdfile in the INPUT. So this confirms stackoverflow.

I don't know this is only required if the mdfile is outside the source path (which was the case in all my test cases) or also if inside the source path,

Reply:
Okay, in that case I would have expected `os.path.join(source_path, conf.get('use_mdfile_as_mainpage'))`, the space is throwing me off. Based on the naming of the conf option I thought it would be a boolean.

Reply:
Incase of an `use_mdfile_as_mainpage` the input should both contain the package dir aka `source_path` and the path of the mdfile. These need to be split by a space.

The question remains if the path of the mdfile should be converted to an absolute path. Right now both relative and absolute paths are ok, as  doxygen is called in the package dir, but to be robust an absolute path would be better. I will implement this.